2-copy-of-code/lesson-15/scripts/checkout.js

```diff
@@ -1,5 +1,7 @@
+import {renderCheckoutHeader} from './checkout/checkoutHeader.js';
 import {renderOrderSummary} from './checkout/orderSummary.js';
 import {renderPaymentSummary} from './checkout/paymentSummary.js';
 
+renderCheckoutHeader();
```

Comment:
if renderCheckoutHeader() function is called after renderOrderSummary() and renderPaymentSummary(), why doesn't the code work?

Reply:
It should work either way, whether it's being called before or after renderOrderSummary() & renderPaymentSummary(). Make sure to
1.  Give the "checkout-header" <div> element the class "js-checkout-header", so that you can access it in your JS checkoutHeader.js file. (checkout.html)
2. Select this element using the querySelector function and edit its innerHTML property, giving it the HTML code you copied from checkout.html (checkoutHeader.js)

Hope this helps you solve your problem :)

Reply:
After following all the steps, my checkout header does not work and I'm getting the following errors in the console. Any ideas?

TypeError: Cannot set properties of null (setting 'innerHTML')
    at updateCartQuantity (orderSummary.js:139:18)
    at renderOrderSummary (orderSummary.js:142:3)
    at checkout.js:5:1

Reply:
You nees to call rendercheckoutHeader first because the checkoutHeader now contains the headercode and the ordersummary.js needs the code to exist so it can run properly.

this is form ordersummary to update the items quantity
`function updateCartQuantity() {
    const cartQuantity = calculateCartQuantity();
    document.querySelector(
      ".js-current-items"
    ).innerHTML = `${cartQuantity} items`;
  }`

and checkoutHeader.j holds the html (js-current-items) as seen below
`  <div class="checkout-header-middle-section">
          Checkout (<a
            class="return-to-home-link js-current-items"
            href="amazon.html">
            </a>)
  </div>`;`
Hope this helps

Reply:
Hello. Thank you for your reply. It worked now, although I'm not sure whether that was the issue. Actually, my code was a little different than in the tutorial, and I had copied it verbatim from the exercise solution, and apparently there might have been a mismatch.

Reply:
i am facing similar problem were after completing every code the order summary is invisible.what should i do

Reply:
> Hello. Thank you for your reply. It worked now, although I'm not sure whether that was the issue. Actually, my code was a little different than in the tutorial, and I had copied it verbatim from the exercise solution, and apparently there might have been a mismatch.

can you explain how it worked

Reply:
What do you mean 'invisible', have you already imported the renderOrderSummary module on you checkout.js?
- import { renderOrderSummary } from "./checkout/orderSummary.js";
- then use the renderOrderSummary ( )